testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jca/statistics/xa/XaDataSourcePoolStatisticsTestCase.java

```diff
@@ -75,13 +81,25 @@
 
     @Deployment
     public static Archive<?> deploy() {
+        // This ugly hack is needed because of the following line on TxTestUtil
+        // return CDI.current().select(TransactionCheckerSingleton.class).get();
+        final String jbossHome = System.getProperty("jboss.dist");
+        final String modulesDir = jbossHome + "/modules/system/layers/base/org/jboss/as/weld/main/";
+        final File weldModuleFile = Paths.get(modulesDir).toFile().listFiles(pathname ->
+                pathname.getName().startsWith("wildfly-weld")
+                        && pathname.getName().endsWith("-SNAPSHOT.jar"))[0];
+
         final EnterpriseArchive ear = ShrinkWrap.create(EnterpriseArchive.class, APP_NAME + ".ear");
         JavaArchive jar = ShrinkWrap.create(JavaArchive.class, ARCHIVE_NAME + ".jar");
         jar.addClasses(TestEntity.class, SLSB1.class, SLSB.class, TimeoutUtil.class);
         jar.addPackage(TxTestUtil.class.getPackage());
         jar.addAsManifestResource(XaDataSourcePoolStatisticsTestCase.class.getPackage(), "persistence.xml", "persistence.xml");
         ear.addAsModule(jar);
         ear.addAsManifestResource(new StringAsset("Dependencies: com.h2database.h2\n"), "MANIFEST.MF");
+        ear.addAsManifestResource(PermissionUtils.createPermissionsXmlAsset(
+                new FilePermission(weldModuleFile.getAbsolutePath(), "read"),
+                new PropertyPermission("ts.timeout.factor", "read")
+        ), "permissions.xml");
```

Comment:
Can TxTestUtil.timeoutWaitTime_ms be lazy calculated when needed, instead of being calculated in a static initializer? I believe that may eliminate some of the need for these perms, since not all uses of TxTestUtil invoke waitForTimeout.